crates/oxc_linter/src/rules/eslint/no_useless_call.rs

```diff
@@ -0,0 +1,177 @@
+    /// foo.call(undefined, 1, 2, 3);
+    /// foo.apply(undefined, [1, 2, 3]);
+    /// foo.call(null, 1, 2, 3);
+    /// foo.apply(null, [1, 2, 3]);
+    /// 
+    /// // These are same as `obj.foo(1, 2, 3);`
+    /// obj.foo.call(obj, 1, 2, 3);
+    /// obj.foo.apply(obj, [1, 2, 3]);
+    /// 
+    /// // success
+    /// // The `this` binding is different.
+    /// foo.call(obj, 1, 2, 3);
+    /// foo.apply(obj, [1, 2, 3]);
+    /// obj.foo.call(null, 1, 2, 3);
+    /// obj.foo.apply(null, [1, 2, 3]);
+    /// obj.foo.call(otherObj, 1, 2, 3);
+    /// obj.foo.apply(otherObj, [1, 2, 3]);
+    ///
+    /// // The argument list is variadic.
+    /// // Those are warned by the `prefer-spread` rule.
+    /// foo.apply(undefined, args);
+    /// foo.apply(null, args);
+    /// obj.foo.apply(obj, args);
+    /// 
+    /// ```
+    NoUselessCall,
+    suspicious,
+);
+
+fn no_useless_call_diagnostic(span1: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Disallow the use of undeclared variables.")
+        .with_label(span1)
+}
+
+fn is_array_argument(_expr: Option<&Argument>) -> bool {
+    let Some(expr) = _expr else {
+        return false;
+    };
+
+    expr.is_array()
+}
+
+fn is_call_or_non_variadic_apply(call_expr: &CallExpression) -> bool {
+    let Some(member_expr) = skip_chain_expression(&call_expr.callee) else {
+        return false;
+    };
+
+    let Some(static_name) = member_expr.static_property_name() else {
+        return false;
+    };
+
+    (static_name == "call" && call_expr.arguments.len() >= 1) || 
+    (static_name == "apply" && call_expr.arguments.len() == 2 && is_array_argument(call_expr.arguments.get(1)))
+}
+
+fn is_validate_this_arg(ctx: &LintContext, expected_this: Option<&Expression>, this_arg: &Expression) -> bool {
+    match expected_this {
+        Some(expected_this) => {
+            is_same_reference(expected_this.without_parenthesized(), this_arg.without_parenthesized(), ctx)
+        }
+        None => {
+            this_arg.is_null_or_undefined()
+        }
+    }
+}
+
+impl Rule for NoUselessCall {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+
+        if !is_call_or_non_variadic_apply(call_expr) {
+            return;
+        }
+        let Some(callee_member) = skip_chain_expression(&call_expr.callee) else {
+            return;
+        };
+        let expected_this = match skip_chain_expression(callee_member.object()) {
+            Some(member) => Some(member.object()),
+            None => None,
+        };
+        let Some(this_arg) = call_expr.arguments.get(0) else {
+            return;
+        };
+        let Some(this_expr) = this_arg.as_expression() else {
+            return;
+        };
+
+        if is_validate_this_arg(ctx, expected_this, this_expr) {
+            ctx.diagnostic(no_useless_call_diagnostic(call_expr.callee.span()))
```

Comment:
Can we add a fixer for this rule?

Reply:
It seems to we can't fix the expression what had the chain expression.

 And the fix way is from `foo.call(undefined, 1, 2);` -> `foo(1, 2);` ?